crypto/crmf/crmf_lib.c

```diff
@@ -651,9 +651,10 @@ X509
                              encKey->data, encKey->length) <= 0
                 || (ek = OPENSSL_malloc(eksize)) == NULL)
             goto end;
+        ERR_set_mark();
         retval = EVP_PKEY_decrypt(pkctx, ek, &eksize,
                                   encKey->data, encKey->length);
-        ERR_clear_error(); /* error state may have sensitive information */
+        ERR_pop_to_mark(); /* error state may have sensitive information */
```

Comment:
This is not a constant time operation.

Reply:
I do not see why this could be problem here.
Using `ERR_clear_error()` was too aggressive since it deletes all errors.
And using `ERR_set_mark();` ... `ERR_pop_to_mark();` is the usual pattern to get rid of any queue entries potentially added by a function call.
If there is a more efficient way of achieving this please let me know.

Reply:
Please note that the following code uses constant time operation. And ERR_clear_error() is constant time unless allocated err data is present.

Reply:
Ah, now I see what you are after.

Hmm, looks like even `ERR_clear_error()` is not really constant-time due to `OPENSSL_free(es->err_file[i])`, where `es->err_file[i]` should be NULL if and only if no entry is present at index `i`.

And I doubt that constant-time behavior is relevant here because at this point a necrypted newly enrolled certificate has been received and is being decrypted. This only happens under the control of the client requesting a certficate and does not happen often.

Reply:
The `constant_time_*` calls go back to
https://github.com/openssl/openssl/pull/9774#discussion_r321224326

If you prefer, I could move the `ERR_pop_to_mark();` after those.

Reply:
So perhaps @bernd-edlinger should provide some input here.

Reply:
I agree.

Reply:
Tentatively removed the `constant_time_*` calls.

Reply:
According to the exchange on https://github.com/openssl/openssl/issues/17319#issuecomment-998848483 it was fine to remove the `onstant_time_*` calls.

Reply:
Please remove the set_mark/pop_to_mark as well.

Reply:
Adding the set_mark/pop_to_mark was the actual point of that commit.
As explained, using `ERR_clear_error()` was too harsh.
We should remove only those entries that were (potentially) added by `EVP_PKEY_decrypt()`.

Reply:
I mean there is no point in clearing anything as there should be nothing security sensitive in error queue under the same assumptions that lead to removing the consttime calls.

Reply:
Ah, right, so we can simply get rid also of the `ERR_clear_error()` - even better!
Done so, rebasing and squshing the changes to `crmf_lib.c`.

Reply:
According to the ongoing exchange on #17319,
I've moved the changes to `crmf_lib.c` form this PR to an independent one: #17354.